Zend/zend_execute_API.c

```diff
@@ -1417,15 +1481,89 @@ static void zend_set_timeout_ex(zend_long seconds, bool reset_signals) /* {{{ */
 }
 /* }}} */
 
-void zend_set_timeout(zend_long seconds, bool reset_signals) /* {{{ */
+static void zend_set_wall_timeout_ex(zend_long seconds, bool reset_signals) /* {{{ */
 {
+#ifdef ZEND_WIN32
+	zend_executor_globals *eg;
+
+	if (!seconds) {
+		return;
+	}
+
+	/* Don't use ChangeTimerQueueTimer() as it will not restart an expired
```

Comment:
I also didn't bother with the implementation for Windows yet. Since it, among a few other systems, already measures the real execution time, there is not a really good use of the new ini setting on them. Even more so, because a process can only have one of each of the timers (at least in case of `setitimer()`. I'm not sure about `CreateTimerQueueTimer()`).

Currently, I'd lean towards resetting the counter for `max_execution_time` when the `max_execution_wall_time` is modified, but I'm curious about other possibilities.